### PR TITLE
Triggers

### PR DESCRIPTION
**Pull request by helgatheviking**
Rough overhaul that reduces your template overrides to the bare minimum and uses triggers for JS instead of duplicating entire MNM script.

### CODE REVIEW

File: templates/single-product/mnm/mnm-product-quantity.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	ob_start();
+	woocommerce_quantity_input( array(
+		'input_name'  => 'mnm_quantity[' . $mnm_id . ']',
+		'input_value' => $quantity,
+		'min_value'   => $product->get_child_quantity( 'min', $mnm_id ),
+		'max_value'   => $product->get_child_quantity( 'max', $mnm_id ),
+	) );
+
+	$output = ob_get_clean();
+
```
Comment (seb86) [inline templates/single-product/mnm/mnm-product-quantity.php]: @helgatheviking Why is there a lot of `str_replace` happening here?

Comment (helgatheviking) [inline templates/single-product/mnm/mnm-product-quantity.php]: It's in lieu of overriding the `global/quantity-input.php` template itself. I'm adding your read only attribute, and your up/down divs, and then MNM always was swapping the class.

**Comment by helgatheviking** [comment]
@seb86 salut seb. did you get a chance to look at this yet?

**Comment by seb86** [comment]
@helgatheviking I did have a look last week and some changes needed to be applied back in order for it to still work in the Javascript.

I also dealt with the Twenty Sixteen issue with a little CSS.

I will have more time on the weekend to put everything together for a PR.

File: woocommerce-match-box.php [hunk 2]
```diff
@@ -172,7 +172,7 @@ public function plugin_meta_links( $links, $file, $data, $status ) {
 		 */
 		public function includes() {
 			if ( ! is_admin() ) {
-				require_once( 'includes/class-wcmb-template.php' );
+				$this->template = require_once( 'includes/class-wcmb-template.php' );
```
Comment (seb86) [inline woocommerce-match-box.php]: @helgatheviking Where does it say the template was undeclared? I had it return the class like so at the bottom of the file.

`return new WC_Match_Box_Display();`

**Pull request merged**